shell.nix

```diff
@@ -0,0 +1,50 @@
+let
```

Comment:
Please add a comment at the top of the file which provides a detailed explanation of what the purpose of this file is. Assume no knowledge of what nix-shell or nix is.

Reply:
I added a comment, I hope that's descriptive enough :+1:

Reply:
Thanks. But since I don't know Nix is, this doesn't really explain anything to me. However it looks like it's much more useful now for people who do directly intend to use it. Adding a description of why this file exists, assuming no knowledge of Nix, would be important. Thank you!

Reply:
I asked TrueDoctor for some Input and added a bit to the description :+1: